### PR TITLE
Initial drop of unit testcase for Wi-Fi Manager APIs

### PR DESCRIPTION
**Pull request by davidfather**
Unit testcase for Wi-Fi Manager APIs (init & deinit only)

### CODE REVIEW

**Comment by jeongchanKim** [comment]
There are two different kind of changes. I think split to two commit is better

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+void ocftest_wifi_softap_sta_leave(void);
+
+static wifi_manager_cb_s wifi_callbacks = {
+	NULL,						//ocftest_wifi_sta_connected,
+	NULL,						//ocftest_wifi_sta_disconnected,
+	NULL,						//ocftest_wifi_softap_sta_join,
+	NULL,						//ocftest_wifi_softap_sta_leave,
+};
+
+void utc_wifi_manager_init_n(void)
```
Comment (jeongchanKim) [inline apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: all utc functions are used in this file, so let's change to static

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	total_fail = 0;
+
+	utc_wifi_manager_init_n();
+	utc_wifi_manager_init_p();
+
+	utc_wifi_manager_deinit_p();
+	utc_wifi_manager_deinit_n(); // Should be run after positive tc
+
+	utc_wifi_manager_set_mode_n();
+	utc_wifi_manager_set_mode_p();
```
Comment (jeongchanKim) [inline apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: in wifi_manager_set_mode(), there is a memory allocation : wifi_utils_start_soft_ap.
I think before finishing wifi_manager_utc, we should free those utils

File: apps/examples/testcase/ta_tc/wifi_manager/utc/Make.defs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+
+ifeq ($(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_UTC),y)
+CSRCS += utc_wifi_manager_main.c
+
+# Include system io build support
```
Comment (jeongchanKim) [inline apps/examples/testcase/ta_tc/wifi_manager/utc/Make.defs]: let's remove this line

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
```
Comment (jeongchanKim) [inline apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: please add belows
		printf("\n=== TINYARA WIFI_MANAGER TC COMPLETE ===\n");
		printf("\t\tTotal pass : %d\n\t\tTotal fail : %d\n", total_pass, total_fail);

File: apps/examples/testcase/ta_tc/wifi_manager/utc/Make.defs (hunk 3 above)
Comment (davidfather) [reply apps/examples/testcase/ta_tc/wifi_manager/utc/Make.defs]: Done

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c (hunk 1 above)
Comment (davidfather) [reply apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: Good recommendation!. thanks.

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c (hunk 2 above)
Comment (davidfather) [reply apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: Inside wifi_utils_start_soft_ap(), two malloced variables are released before the fuction returns.
Please double-check this.

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c (hunk 4 above)
Comment (davidfather) [reply apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: Thanks for this guide.

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+	ret = wifi_manager_deinit();
+
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_DEINITIALIZED);
+	TC_SUCCESS_RESULT();
+}
+
+int wifi_manager_utc(int argc, FAR char *argv[])
+{
+	total_pass = 0;
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: 1. If you didn't use semaphore, next tc will be launched before finishing this. Let's use it like belows.
2. We should print the result of tc at end of function.
```
int kernel_tc_main(int argc, char *argv[])
{
	sem_wait(&tc_sem);
	working_tc++;

	total_pass = 0;
	total_fail = 0;

	printf("=== TINYARA Kernel TC START! ===\n");
...
...
	printf("\n=== TINYARA Kernel TC COMPLETE ===\n");
	printf("\t\tTotal pass : %d\n\t\tTotal fail : %d\n", total_pass, total_fail);

	working_tc--;
	sem_post(&tc_sem);

	return 0;
}
```

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int utc_wifi_manager_main(int argc, char *argv[])
+#endif
+{
+#ifdef CONFIG_TASH
+	int ret = 0;
+	ret = tash_cmd_install("wifi_mgr_utc", wifi_manager_utc, TASH_EXECMD_SYNC);
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: This makes tester should type in a command to lauch this. I think we don't need this.

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c (hunk 5 above)
Comment (davidfather) [reply apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: Thanks for letting me know this. Done!

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c (hunk 6 above)
Comment (davidfather) [reply apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: this tash command registration is like a step which we have to follow. Please look at other testcases.
In addition, this tash command help us to quickly test testcases.

**Comment by davidfather** [comment]
I will split into two separate commits and request a new pull request.

**Pull request closed without merging**